flow/display_list.h

```diff
@@ -834,8 +834,9 @@ class DisplayListBuilder final : public virtual Dispatcher, public SkRefCnt {
     }
   }
   void setMaskBlurFilter(SkBlurStyle style, SkScalar sigma) override {
-    if (mask_sigma_valid(sigma) &&
-        (current_mask_style_ != style || current_mask_sigma_ != sigma)) {
+    if (!mask_sigma_valid(sigma)) {
+      setMaskFilter(nullptr);
```

Comment:
Is this the behaviour we actually want? It seems more intuitive to me that if we pass an invalid sigma we error out and don't alter the state of the DisplayList. If we're doing this, please can we explicitly document this behaviour somewhere?

Reply:
This is the behavior of SkMaskFilter::MakeBlur(). If the sigma fails the tests there, they return nullptr and setting that into the paint resets the mask filter to null.

Reply:
Note that using an SkMaskFilter(..., 0.0) filter has always been equivalent to turning the filter off. That's what was happening in the downstream tests.

Reply:
LGTM, but please add an API comment here detailing this behaviour.